omniscidb/QueryEngine/ResultSetSort.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2023 Intel Corporation
+ * Copyright 2017 MapD Technologies, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include "ResultSet.h"
+
+#include <functional>
+
+using Comparator = std::function<bool(const PermutationIdx, const PermutationIdx)>;
```

Comment:
The name looks generic and may clash. Do we have any good namespace for it to live in?

Reply:
Unfortunately, not. But I can rename it to `ResultSetComparatorFn`